openpype/hosts/blender/plugins/create/create_render.py

```diff
@@ -0,0 +1,302 @@
+        # Set the scene to use the compositor node tree to render
+        bpy.context.scene.use_nodes = True
+
+        tree = bpy.context.scene.node_tree
+
+        # Get the Render Layers node
+        rl_node = None
+        for node in tree.nodes:
+            if node.bl_idname == "CompositorNodeRLayers":
+                rl_node = node
+                break
+
+        # If there's not a Render Layers node, we create it
+        if not rl_node:
+            rl_node = tree.nodes.new("CompositorNodeRLayers")
+
+        # Get the enabled output sockets, that are the active passes for the
+        # render.
+        # We also exclude some layers.
+        exclude_sockets = ["Image", "Alpha"]
+        passes = [
+            socket
+            for socket in rl_node.outputs
+            if socket.enabled and socket.name not in exclude_sockets
+        ]
+
+        # Remove all output nodes
+        for node in tree.nodes:
+            if node.bl_idname == "CompositorNodeOutputFile":
+                tree.nodes.remove(node)
+
+        # Create a new output node
+        output = tree.nodes.new("CompositorNodeOutputFile")
+
+        if ext == "exr" and multilayer:
+            output.layer_slots.clear()
+            filepath = f"{name}{aov_sep}AOVs.####"
+            output.base_path = os.path.join(output_path, filepath)
+        else:
+            output.file_slots.clear()
+            output.base_path = output_path
+
+        image_settings = bpy.context.scene.render.image_settings
+        output.format.file_format = image_settings.file_format
+
+        aov_file_products = []
+
+        # For each active render pass, we add a new socket to the output node
+        # and link it
+        for render_pass in passes:
+            if ext == "exr" and multilayer:
+                output.layer_slots.new(render_pass.name)
+            else:
+                filepath = f"{name}{aov_sep}{render_pass.name}.####"
+
+                output.file_slots.new(filepath)
+
+                aov_file_products.append(
+                    (render_pass.name, os.path.join(output_path, filepath)))
+
+            node_input = output.inputs[-1]
+
+            tree.links.new(render_pass, node_input)
+
+        return aov_file_products
+
+    @staticmethod
+    def imprint_render_settings(node, data):
+        RENDER_DATA = "render_data"
+        if not node.get(RENDER_DATA):
+            node[RENDER_DATA] = {}
+        for key, value in data.items():
+            if value is None:
+                continue
+            node[RENDER_DATA][key] = value
+
+    def prepare_rendering(self, asset_group, name):
+        filepath = bpy.data.filepath
+        assert filepath, "Workfile not saved. Please save the file first."
+
+        file_path = os.path.dirname(filepath)
+        file_name = os.path.basename(filepath)
+        file_name, _ = os.path.splitext(file_name)
+
+        project = get_current_project_name()
+        settings = get_project_settings(project)
```

Comment:
Settings should be retrieved in the `apply_settings` method. Right?

Reply:
I'm just finding out about this method 😅 My bad, I'll change it!

Reply:
Wait! :) I think Blender is still using the legacy publisher and thus this might not work for Blender.

Reply:
Mmmh, I see that the legacy creator has an `apply_settings` method as well, but both require the settings as parameters, so they must be retrieved before anyway! And yeah, I cannot find any old creator that uses is, so for now I'll leave it like this and maybe we can change it when we implement the new publisher in Blender :)